addons/sale_coupon/tests/test_program_numbers.py

```diff
@@ -790,10 +787,30 @@ def test_coupon_and_coupon_discount_fixed_amount_tax_incl(self):
                 'coupon_code': 'test_10pc'
             }).process_coupon()
         order.recompute_coupon_lines()
-        #TODO fix numbers
-        self.assertEqual(order.amount_tax, 9.01)
-        self.assertEqual(order.amount_untaxed, 0.08)
-        self.assertEqual(order.amount_total, 9.09, "The promotion program should not be altered after recomputation")
+        self.assertEqual(order.amount_total, 0.0, "The promotion program should not be altered after recomputation")
+
+    def test_program_discount_percentage_100(self):
+        # test 100% percentage discount (tax included)
+
+        self.env['sale.coupon.program'].create({
+            'name': '100% discount',
+            'promo_code_usage': 'no_code_needed',
+            'program_type': 'promotion_program',
+            'discount_percentage': 100.0,
+            'rule_minimum_amount_tax_inclusion': 'tax_included',
+        })
+        self.tax_10pc_incl.price_include = True
+
+        self.drawerBlack.taxes_id = self.tax_10pc_incl
+        order = self.empty_order
+        order.order_line = self.env['sale.order.line'].create({
+            'product_id': self.drawerBlack.id,
+            'product_uom_qty': 1.0,
+            'order_id': order.id,
+        })
+        order.recompute_coupon_lines()
+        self.assertEqual(len(order.order_line.ids), 2, "discount should be applied")
+        self.assertEqual(order.amount_total, 0.0, "100% discount should be applied")
```

Comment:
Would you be fine adding 3 more lines here to strengthen the test? I feel like using 100% and not checking taxes is a bit light and might let a bug go through, eg having still 0 total but wrong taxes.

Something like:
```
set coupon to 95%
recompute
check tax, subtotal and total
```